athos/methods/auth.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2024 Nethesis S.r.l.
+ * http://www.nethesis.it - info@nethesis.it
+ *
+ * This file is part of Dartagnan project.
+ *
+ * Dartagnan is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or any later version.
+ *
+ * Dartagnan is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Dartagnan.  If not, see COPYING.
+ *
+ */
+
+package methods
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	_ "github.com/jinzhu/gorm/dialects/mysql"
+	"github.com/nethesis/dartagnan/athos/database"
+	"github.com/nethesis/dartagnan/athos/models"
+	"github.com/nethesis/dartagnan/athos/utils"
+)
+
+func BasicAuth(c *gin.Context) {
+	// define model
+	var system models.System
+
+	// get http basic credentials
+	uuid, secret, _ := c.Request.BasicAuth()
+
+	// init db instance
+	db := database.Instance()
+	db.Preload("Subscription.SubscriptionPlan").Where("uuid = ? AND secret = ?", uuid, secret).First(&system)
+
+	// check if system exists
+	if system.ID == 0 {
```

Comment:
For security reasons, we should use constant time compare: https://pkg.go.dev/crypto/subtle#ConstantTimeCompare

An example here: https://www.alexedwards.net/blog/basic-authentication-in-go